cli/tests/test_git_push.rs

```diff
@@ -835,6 +835,28 @@ fn test_git_push_no_description() {
     );
 }
 
+#[test]
+fn test_git_push_no_description_in_immutable() {
+    let (test_env, workspace_root) = set_up();
+    test_env.jj_cmd_ok(&workspace_root, &["branch", "create", "imm"]);
+    test_env.jj_cmd_ok(&workspace_root, &["describe", "-m="]);
+    test_env.add_config(r#"revset-aliases."immutable_heads()" = "imm""#);
```

Comment:
nit: for sanity check, can you add `git push --dry-run` without `immutable_heads() = imm`?

Reply:
Done.